File: serveur/app/models/water_intake.py
# app/models/water_intake.py
#
# Hydratation (2026-07-05). Type A user-scoped.
# Chaque row = une prise d'eau horodatée (un verre / une bouteille). Le total du
# jour se dérive côté client par SUM(amount_ml) sur la date. Le serveur stocke
# les événements bruts, il ne calcule aucun agrégat (décision produit 2026-07-05).
# `date` = jour local "YYYY-MM-DD" (regroupement) ; `created_at` = instant de la
# prise (heure de la journée). Objectif journalier versionné via health_goals
# (type WATER_ML), pas d'entité dédiée.

from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from app.database import Base
import uuid


class WaterIntake(Base):
    __tablename__ = "water_intakes"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    uuid = Column(String, nullable=False, unique=True, default=lambda: str(uuid.uuid4()))

    date = Column(String, nullable=False, index=True)   # "YYYY-MM-DD" (jour local, convention projet)
    amount_ml = Column(Integer, nullable=False)          # volume d'une prise en ml (> 0)

    created_at = Column(DateTime(timezone=True), nullable=True)   # instant de la prise
    updated_at = Column(DateTime(timezone=True), nullable=True)

    def __repr__(self):
        return (
            f"WaterIntake(id={self.id}, user_id={self.user_id}, uuid='{self.uuid}', "
            f"date='{self.date}', amount_ml={self.amount_ml})"
        )

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/WaterIntake.kt
package com.example.sportapp.core.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601

/**
 * Hydratation (2026-07-05) — portage Android du modèle serveur `water_intakes`
 * (cf. serveur/app/models/water_intake.py). Type A user-scoped.
 *
 * Chaque row = une prise d'eau horodatée (un verre / une bouteille). Le total du
 * jour se dérive côté client par SUM(amountMl) sur la date. `date` = jour local
 * "YYYY-MM-DD" (regroupement) ; `createdAt` = instant de la prise (heure de la
 * journée) — préservé au fil des updates (seul `updatedAt` est rafraîchi).
 * Objectif journalier versionné via HealthGoal (type WATER_ML), pas ici.
 */
@Entity(
    tableName = "water_intakes",
    indices = [
        Index(value = ["uuid"], unique = true),
        Index(value = ["user_id"]),
        Index(value = ["date"]),
    ]
)
data class WaterIntake(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,

    val date: String,                                    // "YYYY-MM-DD" (jour local)
    @ColumnInfo(name = "amount_ml") val amountMl: Int,   // volume d'une prise en ml (> 0)

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "created_at") val createdAt: String? = getNowISO8601(),  // instant de la prise
    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
